opentelemetry-appender-tracing/src/layer.rs

```diff
@@ -1,60 +1,122 @@
+    filepath
+}
+
+impl EventVisitor {
+    fn visit_metadata(&mut self, meta: &Metadata) {
+        self.log_record_attributes
+            .push(("log.name".into(), meta.name().into()));
+        self.log_record_attributes
+            .push(("log.target".into(), meta.target().to_owned().into()));
+
+        if let Some(module_path) = meta.module_path() {
+            self.log_record_attributes
+                .push(("log.module.path".into(), module_path.to_owned().into()));
+        }
+        if let Some(filepath) = meta.file() {
+            self.log_record_attributes
+                .push(("log.file.path".into(), filepath.to_owned().into()));
+            self.log_record_attributes.push((
+                "log.file.name".into(),
+                get_filename(filepath).to_owned().into(),
+            ));
+        }
+        if let Some(line) = meta.line() {
+            self.log_record_attributes
+                .push(("log.file.line".into(), line.into()));
+        }
+    }
+
+    fn push_to_otel_log_record(self, log_record: &mut LogRecord) {
+        log_record.body = self.log_record_body;
+        log_record.attributes = Some(self.log_record_attributes);
+    }
+}
+
+impl tracing::field::Visit for EventVisitor {
     fn record_debug(&mut self, field: &tracing::field::Field, value: &dyn std::fmt::Debug) {
-        if field.name() == "message" {
-            self.log_record.body = Some(format!("{value:?}").into());
-        } else if let Some(ref mut vec) = self.log_record.attributes {
-            vec.push((field.name().into(), format!("{value:?}").into()));
-        } else {
-            let vec = vec![(field.name().into(), format!("{value:?}").into())];
-            self.log_record.attributes = Some(vec);
+        if !is_metadata(field.name()) {
+            if field.name() == "message" {
+                self.log_record_body = Some(format!("{value:?}").into());
+            } else {
+                self.log_record_attributes
+                    .push((field.name().into(), format!("{value:?}").into()));
+            }
         }
     }
 
     fn record_str(&mut self, field: &tracing_core::Field, value: &str) {
-        if let Some(ref mut vec) = self.log_record.attributes {
-            vec.push((field.name().into(), value.to_owned().into()));
-        } else {
-            let vec = vec![(field.name().into(), value.to_owned().into())];
-            self.log_record.attributes = Some(vec);
+        if !is_metadata(field.name()) {
+            self.log_record_attributes
+                .push((field.name().into(), value.to_owned().into()));
         }
     }
 
     fn record_bool(&mut self, field: &tracing_core::Field, value: bool) {
-        if let Some(ref mut vec) = self.log_record.attributes {
-            vec.push((field.name().into(), value.into()));
-        } else {
-            let vec = vec![(field.name().into(), value.into())];
-            self.log_record.attributes = Some(vec);
+        if !is_metadata(field.name()) {
```

Comment:
isn't this a visitor for attributes of type `bool`?  Most (all?) of the metadata attributes are not of `type` bool, so isn't searching the list of metadata attributes to see if any of them match the name of this attribute a bit wasteful?  Or, I guess my question is, why do we need to check if this bool attribute is in the set of any metadata attributes that are not bool?  (And the same question for all the other visit methods other than `record_str`...)

Reply:
I think you're right but if I want to be really really pedantic I could say someone could add an attribute of type bool in a log line with one of the metadata. But it seems really really unlikely. :p 

The logline is an Int so we could check only for this metadata in record_i64.